modules/charon-core/src/test/java/org/wso2/charon3/core/utils/PatchOperationUtilTest.java

```diff
@@ -0,0 +1,400 @@
+        PatchOperation patchOperation2 = PatchOperation.class.newInstance();
+
+        patchOperation1.setOperation(remove);
+        patchOperation1.setPath("nickName");
+
+        patchOperation2.setOperation(remove);
+        patchOperation2.setPath("emails[type eq home]");
+
+        User oldResource = getUser();
+        User copyOfOldResource = (User) CopyUtil.deepCopy(oldResource);
+        SCIMResourceTypeSchema scimResourceTypeSchema = getSchema();
+
+        return new Object[][]{
+
+                {patchOperation1, oldResource, copyOfOldResource, scimResourceTypeSchema},
+                {patchOperation2, oldResource, copyOfOldResource, scimResourceTypeSchema}
+        };
+    }
+
+    @Test(dataProvider = "dataForPatchRemoveSuccess")
+    public void testDoPatchRemoveSuccess(PatchOperation operation, AbstractSCIMObject oldResource,
+                                         AbstractSCIMObject copyOfOldResource, SCIMResourceTypeSchema schema)
+            throws BadRequestException, NotImplementedException, CharonException {
+
+        PatchOperationUtil.doPatchRemove(operation, oldResource, copyOfOldResource, schema);
+
+        Assert.assertTrue(true, "doPatchRemove is successful");
+    }
+
+    @DataProvider(name = "dataForPatchRemoveExceptions")
+    public Object[][] dataToPatchRemoveExceptions() throws InstantiationException, IllegalAccessException,
+            CharonException {
+
+        PatchOperation patchOperation1 = PatchOperation.class.newInstance();
+        PatchOperation patchOperation2 = PatchOperation.class.newInstance();
+        PatchOperation patchOperation3 = PatchOperation.class.newInstance();
+
+        patchOperation1.setOperation(remove);
+
+        patchOperation2.setOperation(remove);
+        patchOperation2.setPath("permissions");
+
+        patchOperation3.setOperation(remove);
+        patchOperation3.setPath("emails[type sw home]");
+
+        User oldResource = getUser();
+        User copyOfOldResource = (User) CopyUtil.deepCopy(oldResource);
+        SCIMResourceTypeSchema scimResourceTypeSchema = getSchema();
+
+        return new Object[][]{
+
+                {patchOperation1, oldResource, copyOfOldResource, scimResourceTypeSchema},
+                {patchOperation2, oldResource, copyOfOldResource, scimResourceTypeSchema},
+                {patchOperation3, oldResource, copyOfOldResource, scimResourceTypeSchema}
+        };
+    }
+
+    @Test(dataProvider = "dataForPatchRemoveExceptions",
+            expectedExceptions = {BadRequestException.class, NotImplementedException.class})
+    public void testDoPatchRemoveExceptions(PatchOperation operation, AbstractSCIMObject oldResource,
+                                            AbstractSCIMObject copyOfOldResource, SCIMResourceTypeSchema schema)
+            throws BadRequestException, NotImplementedException, CharonException {
+
+        PatchOperationUtil.doPatchRemove(operation, oldResource, copyOfOldResource, schema);
+
```

Comment:
unnecessary new line

Reply:
fixed in 6e37f6e580f0e7c973edbc994d55b5a8062d5c14